src/azul/service/manifest_service.py

```diff
@@ -776,6 +809,409 @@ def manifest_config(self) -> ManifestConfig:
 Bundles = MutableMapping[FQID, Bundle]
 
 
+@dataclass(frozen=True)
+class Relation:
```

Comment:
This module is getting too big. I think the PFB manifest generator should remain here but the support classes should be in a different module. The generator should act as a slim facade.